### PR TITLE
Fix describe returning oldest matching tag

### PR DESCRIPTION
**Pull request by ThadHouse**
Low level bug that will not be fixed, so this manually parses the tags to figure out the right tag

### CODE REVIEW

**Comment by ThadHouse** [comment]
Java 11 does not like the groovy methods with `.` in the name, thats why they were replaced with `_`

File: src/main/groovy/edu/wpi/first/wpilib/versioning/WPILibVersioningPlugin.groovy [hunk 1]
```diff
@@ -66,8 +69,41 @@ class WPILibVersioningPlugin implements Plugin<Project> {
...
         Grgit git = Grgit.open(currentDir: (Object)gitDir.absolutePath)
+        // Get the tag given by describe
         String tag = git.describe()
+
+        // Get a list of all tags
+        List<Tag> tags = git.tag.list()
+
+        Tag describeTag = null
+
+        // Find tag hash that starts with describe
```
Comment (333fred) [inline src/main/groovy/edu/wpi/first/wpilib/versioning/WPILibVersioningPlugin.groovy]: Why is this whole rigmarole necessary?

Comment (PeterJohnson) [reply src/main/groovy/edu/wpi/first/wpilib/versioning/WPILibVersioningPlugin.groovy]: Which part?  The describe part?  Or the sort by date part?

Comment (ThadHouse) [reply src/main/groovy/edu/wpi/first/wpilib/versioning/WPILibVersioningPlugin.groovy]: grgit's describe returns the oldest tag attached to a commit if there are multiple tags pointing to the same commit. Not the newest tag.

Comment (333fred) [reply src/main/groovy/edu/wpi/first/wpilib/versioning/WPILibVersioningPlugin.groovy]: Ah, got it. Thanks. Consider an update to put this description in a comment.

Comment (ThadHouse) [reply src/main/groovy/edu/wpi/first/wpilib/versioning/WPILibVersioningPlugin.groovy]: Yeah I'll do that. It just has been causing maven to overwrite builds, so it needed to be fixed.

**Pull request merged**